lib/util/Queue.js

```diff
@@ -9,11 +9,19 @@ var promiseutil = require("../promiseutil");
  * is in its critical section. Only 1 item at any one time will be calling this function.
  * The function should return a Promise which is resolved/rejected when the next item
  * can be taken from the queue.
+ * @param {integer} intervalMs Optional. If provided and > 0, this queue will be serviced
+ * at an interval of intervalMs. Otherwise, items will be processed imediately.
  */
-function Queue(processFn) {
+function Queue(processFn, intervalMs) {
     this._queue = [];
     this._processing = null;
     this._procFn = processFn; // critical section Promise<result> = fn(item)
+    this._intervalMs = intervalMs;
+
+    if (this._intervalMs) {
```

Comment:
You document what you expect this to be (an integer which will queue if > 0), but you don't check it. This is javascript remember: the language which will attempt to back stab you in a heartbeat:
```javascript
> s = -4
-4
> if (s) { console.log("HERRO") }
HERRO
undefined
> s = 0
0
> if (s) { console.log("HERRO") }
undefined
```

Needs more paranoia, it's out to get yoooooou! :imp: